src/hera/workflows/io/_io_mixins.py

```diff
@@ -50,31 +43,12 @@ def _construct_io_from_fields(cls: Type[BaseModel]) -> Iterator[Tuple[str, Field
         yield field, field_info, construct_io_from_annotation(field, annotations[field])
 
 
-class InputMixin(BaseModel):
-    def __new__(cls, **kwargs):
-        if _context.declaring:
-            # Intercept the declaration to avoid validation on the templated strings
-            # We must then turn off declaring mode to be able to "construct" an instance
-            # of the InputMixin subclass.
-            _context.declaring = False
-            instance = cls.construct(**kwargs)
-            _context.declaring = True
-            return instance
-        else:
-            return super(InputMixin, cls).__new__(cls)
-
-    def __init__(self, /, **kwargs):
-        if _context.declaring:
-            # Return in order to skip validation of `construct`ed instance
-            return
-
-        super().__init__(**kwargs)
-
+class InputMixin:
     @classmethod
     def _get_parameters(cls, object_override: Optional[Self] = None) -> List[Parameter]:
         parameters = []
 
-        for field, field_info, param in _construct_io_from_fields(cls):
+        for field, field_info, param in _construct_io_from_fields(cls):  # type: ignore
```

Comment:
nit: I think I would slightly prefer `cast` over `ignore`s

```suggestion
        for field, field_info, param in _construct_io_from_fields(cast(type[BaseModel], cls)):
```

(Same for the other ignores)